torchtext/experimental/datasets/raw/translation.py

```diff
@@ -0,0 +1,391 @@
+        self._src_iterator = src_iterator
+        self._tgt_iterator = tgt_iterator
+        self.has_setup = False
+        self.start = 0
+        self.num_lines = None
+
+    def setup_iter(self, start=0, num_lines=None):
+        self.start = start
+        self.num_lines = num_lines
+        self.has_setup = True
+
+    def __iter__(self):
+        if not self.has_setup:
+            self.setup_iter()
+
+        for i, item in enumerate(zip(self._src_iterator, self._tgt_iterator)):
+            if i >= self.start:
+                yield item
+            if (self.num_lines is not None) and (i == (self.start +
+                                                       self.num_lines)):
+                break
+
+    def get_iterator(self):
+        return self._iterator
+
+
+def Multi30k(languages="de-en",
+             train_filename="train",
+             valid_filename="val",
+             test_filename="test_2016_flickr",
+             root='.data'):
+    """ Define translation datasets: Multi30k
+        Separately returns train/valid/test datasets as a tuple
+
+    Arguments:
+        languages: The source and target languages for the datasets.
+            Will be used as a suffix for train_filename, valid_filename,
+            and test_filename. The first split (before -) is the source
+            and the second split is the target.
+            Default: 'de-en' for source-target languages.
+        train_filename: The source and target filenames for training
+            without the extension since it's already handled by languages
+            parameter.
+            Default: 'train'
+        valid_filename: The source and target filenames for valid
+            without the extension since it's already handled by languages
+            parameter.
+            Default: 'val'
+        test_filename: The source and target filenames for test
+            without the extension since it's already handled by languages
+            parameter.
+            Default: 'test2016'
+        root: Directory where the datasets are saved. Default: ".data"
+
+    Examples:
+        >>> from torchtext.datasets import Multi30k
+        >>> train_dataset, valid_dataset, test_dataset = Multi30k()
+    """
+    return _setup_datasets("Multi30k",
+                           languages=languages,
+                           train_filename=train_filename,
+                           valid_filename=valid_filename,
+                           test_filename=test_filename,
+                           root=root)
+
+
+def IWSLT(languages='de-en',
+          train_filename='train.de-en',
+          valid_filename='IWSLT16.TED.tst2013.de-en',
+          test_filename='IWSLT16.TED.tst2014.de-en',
+          root='.data'):
+    """ Define translation datasets: IWSLT
+        Separately returns train/valid/test datasets
+        The available datasets include:
+
+    Arguments:
+        languages: The source and target languages for the datasets.
```

Comment:
Same as previous comments

Reply:
fixed, please let me know if it's sufficient